### ISSUE TITLE
softmax的结果 sum 不等于1

### ISSUE THREAD

**Issue by ZhiluDing** [PaddlePaddle/PaddleClas#1478]
PaddleClas版本 release/2.3
基于GeneralRecognition_PPLCNet_x2_5.yaml的配置以及官方给的预训练模型训练了自己的一个模型，训练的时候是个27类，相关主要配置如下：
```yaml
Global:
  checkpoints: null
  pretrained_model: ./deploy/retrieval_models/general_pre_train/best
  output_dir: ./output/crawl
  device: gpu
  save_interval: 1
  eval_during_train: True
  eval_interval: 1
  epochs: 100
  print_batch_step: 10
  use_visualdl: False
  # used for static mode and model export
  image_shape: [3, 224, 224]
  save_inference_dir: ./inference
  eval_mode: retrieval
  use_dali: False
  to_static: False

# model architecture
Arch:
  name: RecModel
  infer_output_key: logits
  infer_add_softmax: True

  Backbone:
    name: PPLCNet_x2_5
    pretrained: True
    use_ssld: True
  BackboneStopLayer:
    name: flatten_0
  Neck:
    name: FC
    embedding_size: 1280
    class_num: 512
  Head:
     name: ArcMargin
     embedding_size: 512
     class_num: 27
     margin: 0.2
     scale: 30
Loss:
  Train:
    - CELoss:
        weight: 1.0
  Eval:
    - CELoss:
        weight: 1.0

Optimizer:
  name: Momentum
  momentum: 0.9
  lr:
    name: Cosine
    learning_rate: 0.04
    warmup_epoch: 5
  regularizer:
    name: 'L2'
    coeff: 0.00001
```

infer_output_key: logits 和 infer_add_softmax: True 都修改成这样了，想看以下最后logits的结果和softmax如何。但是当我使用 export将模型导出成 inference 模型之后，再使用predict_clas.py 去看最后的结果，发现 softmax 的和 不等于1，觉得很奇怪。
perdict_clas.py 里面的代码改成下面：
```python
  def main(config):
      rec_predictor = RecPredictor(config)
      image_list = get_image_list(config["Global"]["infer_imgs"])
  
      batch_imgs = []
      batch_names = []
      cnt = 0
      for idx, img_path in enumerate(image_list):
          img = cv2.imread(img_path)
          print('img_path', img_path)
          if img is None:
              logger.warning(
                  "Image file failed to read and has been skipped. The path: {}".
                  format(img_path))
          else:
              img = img[:, :, ::-1]
              batch_imgs.append(img)
              img_name = os.path.basename(img_path)
              batch_names.append(img_name)
              cnt += 1
  
          if cnt % config["Global"]["batch_size"] == 0 or (idx + 1) == len(image_list):
              if len(batch_imgs) == 0: 
                  continue
                  
              batch_results = rec_predictor.predict(batch_imgs)
              for number, result_dict in enumerate(batch_results):
                  filename = batch_names[number]
                  # print("{}:\t{}".format(filename, result_dict))
                  print('result_dict shape', result_dict.shape)
                  print('result_dict', result_dict)
                  print('sum result', sum(result_dict))
              batch_imgs = []
              batch_names = []
  
      return
```

结果是这样的
```shell
img_path ./dataset/crawl/train/26\311637313711.8573453.jpg
output_names ['save_infer_model/scale_0.tmp_1']
output_tensor <paddle.fluid.core_avx.PaddleInferTensor object at 0x0000017DFC7F2260>
result_dict shape (27,)
result_dict [0.18922791 0.19217324 0.1829695  0.15234679 0.19042316 0.14705104
 0.17601342 0.23064235 0.18011475 0.16751501 0.18701936 0.16641027
 0.17572714 0.2090446  0.15025532 0.17611966 0.16864435 0.17920245
 0.17197414 0.2080395  0.21135774 0.17495109 0.16254929 0.16374113
 0.19615729 0.22133528 0.35745952]
sum result 5.088465303182602
```

我查看了export 之后 的 pdmodel 的 netron 结构，确实是过了 softmax 的，可是这个结果 sum ！= 1 就很奇怪求帮助
![20211124184533](https://user-images.githubusercontent.com/27245627/143224526-9c832dd3-32b2-46c6-9bc6-b911355405c1.png)

**Comment by RainFrost1** [comment]
能够提供一下相应的pretrained_model，我们验证一下 @ZhiluDing

**Comment by ZhiluDing** [comment]
best.pdparams 文件有 30多M，貌似传不上来，有其他方式给你吗？ @RainFrost1

**Comment by RainFrost1** [comment]
百度网盘分享一下？ @ZhiluDing

**Comment by ZhiluDing** [comment]
链接：https://pan.baidu.com/s/1Hi96kxDjv5XcMfiySBilcQ 
提取码：0qy4
@RainFrost1

**Comment by RainFrost1** [comment]
好的，我下周测试一下，感谢~

**Comment by RainFrost1** [comment]
我用模型测试了一下，使用`predict_cls.py`，`sum`结果是等于1的。看你的用的代码，应该是用的`predict_rec.py`脚本，此脚本在模型输出后，会对feature做`l2norm`，所以使用此脚本，`sum`结果可能不等于1

**Comment by ZhiluDing** [comment]
@RainFrost1 明白了，我去试一下。
